frontend/src/api/github.ts

```diff
@@ -104,6 +104,35 @@ export const retrieveGitHubUser = async () => {
   return user;
 };
 
+export const searchPublicRepositories = async (
+  query: string,
+  per_page = 5,
+  sort: "" | "updated" | "stars" | "forks" = "stars",
+  order: "desc" | "asc" = "desc",
+): Promise<GitHubRepository[]> => {
+  const sanitizedQuery = query.trim();
+  if (!sanitizedQuery) {
+    return [];
+  }
+
+  try {
+    const response = await github.get<{ items: GitHubRepository[] }>(
+      "/search/repositories",
+      {
+        params: {
+          q: sanitizedQuery,
+          per_page,
+          sort,
+          order,
+        },
+      },
+    );
+    return response.data.items;
+  } catch (error) {
+    return [];
+  }
```

Comment:
If/when you are going to outsource the logic in `frontend/src/components/features/github/github-repo-selector.tsx` into a custom hook with TanStack Query, you should let this throw. That way we can have better error handling via the query/mutation hooks (see https://tanstack.com/query/latest/docs/framework/react/guides/query-functions)